src/Async-Call.ts

```diff
@@ -62,6 +62,7 @@ export function AsyncCall<OtherSideImplementedFunctions = {}>(
     let isThisSideImplementationPending = true
     let resolvedThisSideImplementationValue: unknown = undefined
     let rejectedThisSideImplementation: unknown = undefined
+    let resolvedChannel: CallbackBasedChannel | EventBasedChannel | null = null
```

Comment:
please record why the channel fails, and re-throw the same error object for every future API call in the future.